curricula/models.py

```diff
@@ -104,7 +105,12 @@ def get_absolute_url(self):
         return reverse('curriculum:curriculum_view', args=[self.slug])
 
     def get_pdf_url(self):
-        return reverse('curriculum:curriculum_pdf', args=[self.slug])
+        url = reverse('curriculum:curriculum_pdf', args=[self.slug])
+        language = get_language()
+        if (language != 'en-us' and language not in settings.LANGUAGE_GENERATE_PDF):
+            m = re.match(r'(/[^/]*)(/.*$)', url)
+            url = m.groups()[1]
+        return url
```

Comment:
would it be feasible to instead do something like

```python
def get_pdf_url(self):
  language = get_language()
  if language != settings.LANGUAGE_CODE and language not in settings.LANGUAGE_GENERATE_PDF:
    language = settings.LANGUAGE_CODE
  with translation.override(language):
    return reverse('curriculum:curriculum_pdf', args=[self.slug])
```

Reply:
oh that's much better!